src/js/game/systems/item_processor.js

```diff
@@ -572,23 +573,23 @@ export class ItemProcessorSystem extends GameSystemWithFilter {
         const item = payload.items[0].item;
         const now = this.root.time.now();
 
+        goalComp.currentDeliveredItems = Math.min(
+            goalComp.currentDeliveredItems + 1,
+            globalConfig.goalAcceptorItemsRequired
+        );
+
+        if (goalComp.item && !item.equals(goalComp.item)) {
+            goalComp.clearItems();
+        }
```

Comment:
Why not put the first chunk in an `else` after the second chunk?

Reply:
yeah will do, thanks :)